dynalab_cli/test.py

```diff
@@ -64,132 +83,155 @@ def run_command(self):
 
         if self.args.local:
             self.run_local_test(config)
-
         else:
-            tmp_dir = os.path.join(self.config_handler.config_dir, "tmp")
-            os.makedirs(tmp_dir, exist_ok=True)
-            # tarball everything
-            print("Tarballing the project directory...")
-            exclude_list_file = os.path.join(tmp_dir, "exclude.txt")
-            self.config_handler.write_exclude_filelist(
-                exclude_list_file, self.args.name, exclude_model=True
-            )
-            process = subprocess.run(
-                [
-                    "tar",
-                    f"--exclude-from={exclude_list_file}",
-                    "-czf",
-                    os.path.join(tmp_dir, f"{self.args.name}.tar.gz"),
-                    ".",
-                ],
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-                universal_newlines=True,
-            )
-            if process.returncode != 0:
-                raise RuntimeError(
-                    f"Exception in tarballing the project directory {process.stderr}"
-                )
-
-            # torch model archive
-            print("Archive the model for torchserve...")
-            archive_command = [
-                "torch-model-archiver",
-                "--model-name",
-                self.args.name,
-                "--serialized-file",
-                config["checkpoint"],
-                "--handler",
-                config["handler"],
-                "--version",
-                "1.0",
-                "-f",
-                "--export-path",
-                tmp_dir,
-            ]
-            if config["model_files"]:
-                archive_command += ["--extra-files", ",".join(config["model_files"])]
-            process = subprocess.run(
-                archive_command,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-                universal_newlines=True,
-            )
-            if process.returncode != 0:
-                raise RuntimeError(f"Exception in torchserve archive {process.stderr}")
-
-            # pull docker
-            lib_dir = os.path.join(
-                os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
-                "dynalab",
-                "dockerfiles",
-            )
-            docker_path = os.path.join(tmp_dir, "Dockerfile")
-            # TODO: pull the files from dynalab repo once public
-            shutil.copyfile(os.path.join(lib_dir, "Dockerfile.dev"), docker_path)
-            shutil.copyfile(
-                os.path.join(lib_dir, "dev-docker-entrypoint.sh"),
-                os.path.join(tmp_dir, "dev-docker-entrypoint.sh"),
+            self.run_docker_test(config)
+
+    def run_docker_test(self, config):
```

Comment:
this diff is pretty bad. I just split the `run_command` in two functions, and de-indented.